kubernetes/resources_api/base/deployment.yaml

```diff
@@ -0,0 +1,91 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: resources-api
+spec:
+  replicas: 2
+  revisionHistoryLimit: 5
+  template:
+    spec:
+      containers:
+      - name: app
+        image: #operationcode/operationcode_backend:latest
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 3000
+        env:
+        - name: FLASK_APP
+          value: run.py
+        - name: SQLALCHEMY_DATABASE_URI
+          valueFrom:
+            secretKeyRef:
+              name: resources-secrets
+              key: sqlalchemy_database_uri
+        - name: POSTGRES_USER
```

Comment:
DB user can go either way. It can be helpful to NOT be a secret, for context, but it can also be consider part of the credential pair. It's a matter of preference.

Reply:
In the `env` file we have the POSTGRES_USER, the USER and the DB_USERNAME all the same. If they were different, I'd be fine making this public, but I didn't think too hard about a world in which they're different

Reply:
My last comment is outdated based on the removal of a bunch of unnecessary environment variables. I can make this public if you like, either way works for me

Reply:
Fine as it is, this comment was mostly for your context. It's a matter of personal preference, and in many cases it's easier to treat them both the same.

As an example, when Hashicorp Vault is involved, I treat both username and password as a single API call, which retrieves an object that has several keys (in this instance it would be`POSTGRES_USER` and` POSTGRES_PASSWORD` and their respective values.)